drivers/storage/port/scsiport/fdo.c

```diff
@@ -705,6 +705,53 @@ FdoHandleDeviceRelations(
     return IoCallDriver(PortExtension->Common.LowerDevice, Irp);
 }
 
+static
+NTSTATUS
+FdoHandleQueryCompatibleId(
+    _Inout_ PIRP Irp)
```

Comment:
Since it seems this function actually acts just on the `Irp->IoStatus.Information` part, why not just make it take an
`_Inout_ PZZWSTR* Ids` (a pointer to a (pointer-to-)double-NULL-terminated string, which is exactly what a MULTI_SZ is) (this allows to have the function self-documented), then you work on that, and you call it like:
`status = FdoHandleQueryCompatibleId((PZZWSTR*)&Irp->IoStatus.Information);`

Reply:
Makes sense, thanks, done.